tests/ui/type-alias-impl-trait/coherence_generalization.rs

```diff
@@ -11,5 +10,6 @@ fn foo<T>() -> Opaque<T> {
 
 impl<T, V> Trait for (T, V, V, u32) {}
 impl<U, V> Trait for (Opaque<U>, V, i32, V) {}
```

Comment:
I'm not sure either what this tests, but I believe we can preserve the original behavior:
```suggestion
impl<T, U, V> Trait for (T, U, V, V, u32) {}
impl<U, V> Trait for (Opaque<U>, U, V, i32, V) {}
```